### ISSUE TITLE
Sourcegraph Extension Missing From Mozilla Extension Repository

### ISSUE THREAD

**Issue by bonedaddy** [sourcegraph/sourcegraph#1346]
Currently the only way to install the extension in firefox is using this url `https://storage.googleapis.com/sourcegraph-for-firefox/latest.xpi`

The extension is not present in the Mozzilla extension shop. It's extremely concerning that the *only* way to do this, is by downloading some random file from a URL that isn't even associated with the extension shop. This is not present on chrome  as can be seen [here](https://chrome.google.com/webstore/detail/sourcegraph/dgjhfomjieaadpoljlnidmbgkdffpack).

Browsing to the mozzile extension [page](https://addons.mozilla.org/en-US/firefox/addon/sourcegraph/versions/1.12.2) gives a 404 not found error.

---

**UPDATE (from Sourcegraph):** The Firefox extension is available to install from https://docs.sourcegraph.com/integration/browser_extension.

**Comment by sqs** [comment]
The ability to run code to support multiple languages requires executing code from blob: URIs. The Mozilla Add-ons site doesn’t permit listing of such extensions (unlike Chrome Web Store, which permits it). Therefore we are self-hosting the extension.  It is signed and published using our and Mozilla’s key, so it is not completely disconnected from Mozilla.

The extension is open source and you can build your own if you’d prefer.

What else could we do (besides listing it on Mozilla Add-ons) to make you feel comfortable?

Cc @ijsnow who can provide additional info.

**Comment by ijsnow** [comment]
We use [Mozilla's `web-ext`tool](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Getting_started_with_web-ext) to [sign](https://github.com/sourcegraph/sourcegraph/blob/master/client/browser/scripts/release-ff.sh#L10) the bundle, so while its not listed on the Addons store, its verified by Mozilla.

As @sqs said, we unfortunately can't use `:blob` as part of the [content security policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP). We tried working with the addon reviewers to get permission to do this but they have a hard line policy against it. as opposed to Chrome which explicitly allows it. We use this to load [Sourcegraph extensions](https://sourcegraph.com/extensions) into a web worker so they are isolated and don't have access to the DOM (as they would if we went with [executeScript](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/tabs/executeScript) as we were suggested) which would introduce more concerns.

Please let us know if you have any more questions or suggestions on what we can do to make you more confident. As sqs mentioned, you can always [view the source code.](https://github.com/sourcegraph/sourcegraph/tree/master/client/browser)

**Comment by postables** [comment]
Ah okay thanks for the explanation that clears things up. @ijsnow Honestly after reading your responses, I have no concerns now. I guess the only thing I would suggest is that on the [page](https://docs.sourcegraph.com/integration/browser_extension) where you give the download link, maybe add a note or something explaining what you both just mentioned?

**Comment by ijsnow** [comment]
Great! Glad that cleared things up. 

@ryan-blunden I propose we add a new page just for the FF extension that explains this and has the download link. It'll be an extra click from the browser extensions page to install, but it'll be the same as Chrome going to the Chrome webstore to download.

**Comment by ryan-blunden** [comment]
Sounds good!

**Comment by sqs** [comment]
The  Firefox extension is back. You can install it from https://docs.sourcegraph.com/integration/browser_extension.

**Comment by doxxx** [comment]
When I try to install the Firefox extension from the docs page, I get the following error popup:

![image](https://user-images.githubusercontent.com/677940/56608913-da857a00-65d9-11e9-908e-9b63e22d8e4c.png)

I tried adding both the docs URL and the XPI storage URL to the exceptions for the "Warn you when websites try to install add-ons" setting, but that had no effect. I also tried turning off that setting entirely, to no effect either.

I also tried disabling the `xpinstall.signatures.required` setting, to no effect.

I'm using Firefox 67.0b12.
